### PR TITLE
fix: add displayCloseButton prop, update storybook

### PR DESCRIPTION
**Pull request by maomaoZH**

### CODE REVIEW

File: packages/storybook-vue/stories/components/modal/Modal.stories.mdx [hunk 1]
```diff
@@ -43,6 +43,13 @@ import ScaleModal from './ScaleModal.vue';
         disable: true,
       },
     },
+    scaleBeforeClose: {
+      description: '@deprecated in v3 in favor of kebab-case event names.',
+    },
+    'scale-before-close': {
+      description:
+        'It fires on every close attempt. Calling `event.preventDefault()` will prevent the modal from closing.',
```
Comment (acstll) [inline packages/storybook-vue/stories/components/modal/Modal.stories.mdx:51]: Would it be possible to add a bit more info? something like this:

`event.detail.trigger` can be one of `CLOSE_BUTTON`, `ESCAPE_KEY` or `BACKDROP`, depending on the action that triggered the event.

File: packages/components/src/components/modal/modal.tsx [hunk 2]
```diff
@@ -59,6 +59,8 @@ export class Modal {
   @Prop() closeButtonLabel?: string = 'Close Pop-up';
   /** (optional) title for close button */
   @Prop() closeButtonTitle?: string = 'Close';
+  /** (optional) display close button */
+  @Prop() displayCloseButton?: boolean = true;
```
Comment (acstll) [inline packages/components/src/components/modal/modal.tsx:63]: What do you think about making the default `false`, thus changing the prop to something like `hideCloseButton` or `omitCloseButton`? I think "boolean" props should be false by default, I would like to have this by convention…

File: packages/components/src/components/modal/modal.tsx [hunk 3]
```diff
@@ -274,18 +276,20 @@ export class Modal {
...
-                part="close-button"
-                onClick={() => this.emitBeforeClose('CLOSE_BUTTON')}
-                aria-label={this.closeButtonLabel}
-                title={this.closeButtonTitle}
-              >
-                <slot name="close-icon">
-                  <scale-icon-action-circle-close decorative />
-                </slot>
-              </button>
+              {this.displayCloseButton && (
```
Comment (acstll) [inline packages/components/src/components/modal/modal.tsx:279]: For some reason the focus trap breaks when the button is not there… let's talk about this 🤓

**Review by acstll — CHANGES_REQUESTED** [review]
More than changes I'm making questions 🙏 — thanks for the PR!

File: packages/storybook-vue/stories/components/modal/Modal.stories.mdx (hunk 1 above)
Comment (maomaoZH) [reply packages/storybook-vue/stories/components/modal/Modal.stories.mdx:51]: updated.

File: packages/components/src/components/modal/modal.tsx (hunk 2 above)
Comment (maomaoZH) [reply packages/components/src/components/modal/modal.tsx:63]: change to `omitCloseButton`

File: packages/components/src/components/modal/modal.tsx (hunk 3 above)
Comment (maomaoZH) [reply packages/components/src/components/modal/modal.tsx]: Hi, i tested again in our `modal.html`, and I also logged the focusable elements. I didnt encounter the issue though. 

https://user-images.githubusercontent.com/23239723/189601625-eace7edb-07f9-4590-825d-e758e400ce98.mov

Comment (acstll) [reply packages/components/src/components/modal/modal.tsx]: I only had the issue in Storybook, but now it's also working there (just tested locally) 🎉 
image

File: packages/storybook-vue/stories/components/modal/Modal.stories.mdx (hunk 1 above)
Comment (acstll) [reply packages/storybook-vue/stories/components/modal/Modal.stories.mdx]: Thank you!

**Comment by acstll** [comment]
All good! 👍

**Pull request merged**